test/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDaoMockTest.java

```diff
@@ -112,219 +116,108 @@ public void giveConsent() {
-            verify(mapper2).load(any());
-            verify(mapper2).delete(any()); // this fails, mapper2 not called
-            verifyNoMoreInteractions(mapper2);
-            verifyNoMoreInteractions(mapper3);
-        }
+        DynamoUserConsent3 consent = captor.getValue();
+        assertTrue(consent.getWithdrewOn() > 0L);
     }
     
     @Test
-    public void withdrawSecondTableThrowsException() {
-        mockMapperResults();
-        doThrow(new ConcurrentModificationException()).when(mapper3).delete(any());
-        
-        // If mapper2 throws exception on delete, resave to mapper3, then propagate exception 
+    public void withdrawConsentWithNoConsent() {
         try {
-            userConsentDao.withdrawConsent(HEALTH_CODE, STUDY_IDENTIFIER);
-            fail("Should have thrown exception");
-        } catch(ConcurrentModificationException e) {
-            verify(mapper2).load(any());
-            verify(mapper2).delete(any()); // throws exception
-            verify(mapper3).query(any(), any());
-            verify(mapper3).delete(any());
-            verifyNoMoreInteractions(mapper2);
-            verifyNoMoreInteractions(mapper3);
+            userConsentDao.withdrawConsent(HEALTH_CODE, STUDY_IDENTIFIER, UNIX_TIMESTAMP);
+            fail("Should have thrown exception.");
+        } catch(BridgeServiceException e) {
+            assertEquals("Consent not found.", e.getMessage());
+            assertEquals(404, e.getStatusCode());
         }
+        verify(mapper).query(any(), any());
+        verifyNoMoreInteractions(mapper);
     }
     
-    @Test
-    public void getConsentFromTable3First() {
-        mockMapperResults();
-        
-        UserConsent consent = userConsentDao.getUserConsent(HEALTH_CODE, STUDY_IDENTIFIER);
-        assertTrue(consent instanceof DynamoUserConsent3);
-        assertEquals(STUDY_IDENTIFIER.getIdentifier(), consent.getStudyIdentifier());
-        
-        verify(mapper3).query(any(), any());
-        verifyNoMoreInteractions(mapper2);
-        verifyNoMoreInteractions(mapper3);
-    }
-
-    @Test
-    public void getConsentFallsbackToTable2() {
-        DynamoUserConsent2 existingConsent2 = new DynamoUserConsent2(HEALTH_CODE, STUDY_IDENTIFIER.getIdentifier());
-        existingConsent2.setConsentCreatedOn(studyConsent.getCreatedOn());
-        existingConsent2.setSignedOn(DateTime.now().getMillis());
-        doReturn(existingConsent2).when(mapper2).load(any());
-
-        UserConsent consent = userConsentDao.getUserConsent(HEALTH_CODE, STUDY_IDENTIFIER);
-        assertTrue(consent instanceof DynamoUserConsent2);
-        assertEquals(STUDY_IDENTIFIER.getIdentifier(), consent.getStudyIdentifier());
-        
-        verify(mapper3).query(any(), any());
-        verify(mapper2).load(any());
-        verifyNoMoreInteractions(mapper2);
-        verifyNoMoreInteractions(mapper3);
-    }
+    // These are basic, more extensive tests can be found in DynamoUserConsentDaoTest.
```

Comment:
I feel like it should be the other way around. Basic tests should be in the non-mock test, to test simple things like reading from/saving to DDB. Semantics and business logic (such as getActiveUserConsent only getting consents that aren't withdrawn) should go in the mock test. This allows us to separate our business logic from our back-end service calls, which in turn makes our code better organized and more robust.